src/software/jetson_nano/thunderloop.cpp

```diff
@@ -147,33 +148,59 @@ void Thunderloop::runLoop()
             }
 
             // If the world msg is new, update the internal buffer
-            // TODO (#2726): check if world has not been sent for a while
             if (new_world.time_sent().epoch_timestamp_seconds() >
                 world_.time_sent().epoch_timestamp_seconds())
             {
+                clock_gettime(CLOCK_MONOTONIC, &last_world_recieved_time);
                 primitive_executor_.updateWorld(new_world);
                 world_ = new_world;
             }
 
+            // If world not received in a while, stop robots
+            struct timespec world_result;
+
+            clock_gettime(CLOCK_MONOTONIC, &current_time);
+            ScopedTimespecTimer::timespecDiff(&current_time, &last_world_recieved_time,
+                                              &world_result);
+
+            auto nanoseconds_elapsed_since_last_world =
+                world_result.tv_sec * static_cast<int>(NANOSECONDS_PER_SECOND) +
+                world_result.tv_nsec;
+
+            if (nanoseconds_elapsed_since_last_world >
+                static_cast<long>(WORLD_TIMEOUT_NS))
+            {
+                primitive_executor_.setStopPrimitive();
+
+                // Log milliseconds since last world received if we are timing out
+                int milliseconds_elapsed_since_last_world = static_cast<int>(
+                    static_cast<double>(world_result.tv_sec) * MILLISECONDS_PER_SECOND +
+                    static_cast<double>(world_result.tv_nsec) *
+                        MILLISECONDS_PER_NANOSECOND);
+
+                LOG(INFO) << "Milliseconds since last world: "
+                          << milliseconds_elapsed_since_last_world;
```

Comment:
Could we update this log to also include the fact that we're overriding the received primitives with StopPrimitive.

Reply:
👍